core/meterer/onchain_state.go

```diff
@@ -0,0 +1,110 @@
+package meterer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/Layr-Labs/eigenda/core/eth"
+)
+
+/* HEAVILY MOCKED */
+
+var (
+	DummyReservationBytesLimit    = uint64(1024)
+	DummyPaymentLimit             = TokenAmount(512)
+	DummyMinimumChargeableSize    = uint32(128)
+	DummyMinimumChargeablePayment = uint32(128)
+
+	DummyReservation     = ActiveReservation{DataRate: DummyReservationBytesLimit, StartTimestamp: 0, EndTimestamp: math.MaxUint32, QuorumSplit: []byte{50, 50}, QuorumNumbers: []uint8{0, 1}}
+	DummyOnDemandPayment = OnDemandPayment{CumulativePayment: DummyPaymentLimit}
```

Comment:
instead of mocking directly in the implementation, the implementation should use `tx` to load these onchain parameters, and the `OnchainPaymentState` used in tests should load a mocked `tx` that returns the mock values

Reply:
moved to use mocked on-chain state, lmk if this is what you had in mind 🤔 I wanted to have some kind of cache of on-chain state in-memory and `tx` updates it periodically, so the focus was just grabbing cached states